pkg/notify/notify.go

```diff
@@ -225,16 +224,17 @@ func deepCopyWithTemplate(value interface{}, tmpl *template.Template, data inter
 
 // toGroupTicketLabel returns the group labels as a single string.
 // This is used to reference each ticket groups.
-// String is the form of an ALERT Prometheus metric name, with all spaces removed.
+// String is the form of ALERT{sha512hash(groupLabels)}
+// hashing ensures that JIRA validation still accepts the output even
+// if the combined length of all groupLabel key-value pairs would be
+// longer than 255 chars
 func toGroupTicketLabel(groupLabels alertmanager.KV) string {
-	buf := bytes.NewBufferString("ALERT{")
+	hash := sha512.New()
 	for _, p := range groupLabels.SortedPairs() {
-		buf.WriteString(p.Name)
-		buf.WriteString(fmt.Sprintf("=%q,", p.Value))
+		kvString := fmt.Sprintf("%s=%q,", p.Name, p.Value)
+		hash.Write([]byte(kvString)) // nolint:errcheck // hash.Write can never return an error
```

Comment:
Well done on nolint and comment 🤗

Reply:
Actually, there might better way! (: 
```suggestion
		_ = hash.Write([]byte(kvString)) 
```